corehq/apps/reports/commtrack/util.py

```diff
@@ -98,8 +98,11 @@ def filter_relevant(queryset):
 
 
 @quickcache(['domain'])
-def get_product_id_name_mapping(domain):
-    return dict(SQLProduct.objects.filter(domain=domain).values_list('product_id', 'name'))
+def get_product_id_name_mapping(domain, product_ids=None):
+    products = SQLProduct.objects.filter(domain=domain)
+    if product_ids:
+        products = products.filter(product_id__in=product_ids)
+    return dict(products.values_list('product_id', 'name'))
```

Comment:
Looks like pagination wouldn't be possible, since this includes only products with ledgers, but you could still move filtering by program into the db call.

Reply:
making the change but dont see an advantage..
One thing i like about the current approach is that it finds product ids and then uses only that for both ES and SQL whereas this divides program id for SQL and product ids for ES. Not sure if that sounds as an advantage but i feel it keeps things simple

Reply:
Plus we still need to find product ids for ES. Open to understanding if there is an advantage with the change.